frontend/public/components/catalog-items.jsx

```diff
@@ -169,26 +209,104 @@ export class CatalogTileViewPage extends React.Component {
     return _.find(allCategories, { id: categoryID }).label;
   }
 
+  filterItems() {
+    const { filters } = this.state;
+    const { byName, byType } = filters;
+    const { items } = this.props;
+
+    let filteredItems = [];
+
+    if (byType.clusterServiceClass.active) {
+      filteredItems = _.filter(items, (item) => item.kind === byType.clusterServiceClass.value);
+    }
+
+    if (byType.imageStream.active) {
+      filteredItems = filteredItems.concat(_.filter(items, (item) => item.kind === byType.imageStream.value));
+    }
+
+    filteredItems = !_.isEmpty(filteredItems) ? filteredItems : items;
```

Comment:
If imageStream is selected an none are image streams, we show all?

Reply:
meant if no filtering by type, then show all (since we start out with no filtered items), then filter byName (if applicable), else no filtering, so return all.  This assumes that we'll always have items of these types, so I see the logic flaw

Reply:
Adjusted the filtering logic